### DISCUSSION TITLE
libp2p-rPGF-1: Nominations for Research & Development

### DISCUSSION THREAD

**Discussion by p-shahi** [libp2p/libp2p#213] [RetroPGF]
### Research & Development

This discussion post is for projects nominations to the Research & Development category.
This category is intended for work on advancing the libp2p specifications and protocols.

### Nomination Template
When nominating a project, please answer the following questions in a comment to this discussion:

```
Project name:
The project name

Description:
A description of the positive impact of this project for this specific category over the past year (March 2023 - March 2024).

Github/Twitter link:
A link to the project’s GitHub or Twitter

Contact:
Contact info for the project or project lead
```

Things to consider when nominating a project:

- Describe how this project has delivered impact to the libp2p ecosystem in the previous year (March 2023 - March 2024). Use the provided template.
- Nominate projects in **one** relevant category. Do not nominate the same project across different categories! If a project has overlap in multiple categories, please choose just one category that best aligns with it. 
- Organizers of libp2p-RPGF-1 may reassign a project’s category from its nominated one if it better aligns elsewhere.
- Check if a project has been nominated already to avoid duplication.
- Please provide feedback to help the process improve in each round. Your input and feedback is valuable and you can use this post to submit feedback now.

Nominations will close on May 3rd, 2024.

See https://github.com/libp2p/libp2p/discussions/209 for an overview of the process, timelines, and to ask general questions.

**Comment by willemolding** [comment]
## Project name:
Libp2p Nym Transport

## Description:
From March 2023 to June 2023 ChainSafe R&D, in collaboration with Nym, conducted research on integrating the Nym mixnet protocol with Libp2p. This was followed up with a prototype that allows libp2p nodes to configure themselves to use this transport and send and receive messages through the mixnet.

See the [announcement article](https://blog.nymtech.net/press-release-nym-and-chainsafe-bring-enhanced-privacy-to-over-half-a-million-ethereum-validators-9792cbdedebd)
and the [full specification](https://hackmd.io/@nZ-twauPRISEa6G9zg3XRw/HkE8sHuns).

The motivation for development was initially to provide network privacy for Ethereum consensus validators, as they use libp2p as their p2p library. However, this transport could be used by any application that uses libp2p, bringing privacy to a wide variety of projects.

The prototype was [integrated with the Lighthouse Ethereum client](https://github.com/ChainSafe/lighthouse/pull/2) laying the groundwork for developing fully private blockchain clients with libp2p.

## Github/Twitter link:
https://github.com/ChainSafe/rust-libp2p-nym
https://twitter.com/ChainSafeth/status/1631729256974753813

## Contact:
Willem Olding (ChainSafe Systems) - R&D Program Lead - willem@chainsafe.io

**Comment by guillaumemichel** [comment]
Project name
---
DHT Reader Privacy Upgrade (aka. Double Hash DHT)

Description
---
Currently in the libp2p DHT, neither readers (clients retrieving content) nor writers (hosts storing and distributing content) have much privacy with regard to content they consume or publish. It is trivial for a DHT server node to associate the requestor's identity with the accessed content during the routing process. A curious DHT server node, can request the same CIDs to find out what content other users are consuming. Improving privacy in the libp2p DHT has been a strong request from the community for some time.

This research project ended up producing a concrete [spec](https://github.com/guillaumemichel/specs/blob/double-hashing-dht/IPIP/0373-double-hash-dht.md) for implementing private provider records in the Amino DHT network, which is made of libp2p DHT nodes. The spec introduces a DHT privacy upgrade boosting the reader’s privacy. It prevents DHT tracking as described above, and add Provider Records Authentication. The proposed modifications also add a slight Writer Privacy improvement as a side effect.

This work was performed by the [ProbeLab](https://probelab.io) team.

### Talks

* https://www.youtube.com/watch?v=ZPIDU1-JnVc
* https://www.youtube.com/watch?v=VBlx-VvIZqU
* https://www.youtube.com/watch?v=m-6_VZ8e1tk
* https://www.youtube.com/watch?v=FP4kKemco4w

Github/Twitter link
---
https://github.com/ipfs/specs/pull/373

Contact
---
@guillaumemichel 
@yiannisbot
